test/tests/functional/pbs_advance_resv_end_hook.py

```diff
@@ -0,0 +1,169 @@
+from tests.functional import *
+import time
+
+
+class TestAdvanceReservationEndHook(TestFunctional):
+    """
+    Tests to verify the reservation end hook for
+    an advanced reservation if it ends or gets deleted
+    """
+
+    def setUp(self):
+        """ Create and import the python script for reservation end hook
+            in setup method"""
+
+        super(TestAdvanceReservationEndHook, self).setUp()
+        hook_script = """
+import pbs
+e=pbs.event()
+
+pbs.logmsg(pbs.LOG_DEBUG, 'Reservation End Hook name - %s' % (pbs.event().hook_name))
+
+if e.type == pbs.RESV_END:
+    pbs.logmsg(pbs.LOG_DEBUG, 'Reservation ID - %s'%pbs.event().resv.resvid)
+
+e.reject()
+"""
+        self.hook_name = 'resvend_hook'
+        attrs = {'event': 'resv_end'}
+        self.server.create_import_hook(
+            self.hook_name, attrs, hook_script, overwrite=True)
+
+        self.server.manager(
+            MGR_CMD_SET, SERVER, {
+                'log_events': 2047}, expect=True)
+
+    def submit_and_confirm_resv(self, offset, duration, select="1:ncpus=1"):
+        """ Submit and confirm the advance reservation """
+
+        start_time = time.time() + offset
+        end_time = start_time + duration
+        attrs = {
+            'reserve_start': start_time,
+            'reserve_end': end_time,
+            'Resource_List.select': select}
+        rid = self.server.submit(Reservation(TEST_USER, attrs))
+
+        resv_attrs = {'reserve_state': (MATCH_RE, 'RESV_CONFIRMED|2')}
+        self.server.expect(RESV, resv_attrs, id=rid)
+
+        return rid
+
+    def test_delete_resv(self):
+        """ Test to delete a confirmed reservation
+            and verify the resvend hook. """
+
+        offset = 3
```

Comment:
23 seconds might be short for some of our test machines.  Consider increasing these variables.